src/logic/currencyValues/store/middleware/index.ts

```diff
@@ -21,14 +23,16 @@ const currencyValuesStorageMiddleware = (store) => (next) => async (action) => {
       case SET_CURRENCY_RATE:
       case SET_CURRENCY_BALANCES: {
         const currencyValues = currencyValuesSelector(state)
-        const currencyValuesWithoutBalances = currencyValues.map((currencyValue) => {
-          const currencyRate = currencyValue.get('currencyRate')
-          const selectedCurrency = currencyValue.get('selectedCurrency')
+
+        const currencyValuesWithoutBalances: Map<string, CurrencyRateValue> = currencyValues.map((currencyValue) => {
+          const currencyRate: number = currencyValue.get('currencyRate')
+          const selectedCurrency: AVAILABLE_CURRENCIES = currencyValue.get('selectedCurrency')
```

Comment:
why do we lose type here?

Reply:
sorry I didn't understand you

Reply:
Ok I think I understand now, since it's a map and immutable.js doesn't infer types for maps